libopae/plugins/xfpga/sysfs.c

```diff
@@ -720,13 +729,26 @@ fpga_result sysfs_get_fme_pwr_path(fpga_token token, char *sysfs_pwr)
 		return FPGA_INVALID_PARAM;
 	}
 	res = cat_token_sysfs_path(sysfs_pwr, token, SYSFS_FORMAT(sysfs_fme_pwr_glob));
+	if (res != FPGA_OK) {
+		return res;
+	}
+
+	// check for path is valid
+	int gres = glob(sysfs_pwr, GLOB_NOSORT, NULL, &pglob);
+	if (gres) {
+		FPGA_ERR("Failed pattern match %s: %s", sysfs_pwr, strerror(errno));
+		globfree(&pglob);
```

Comment:
globfree may fail the underlying buffer is never allocated.

Reply:
fiexd